subsys/bluetooth/host/iso.c

```diff
@@ -2539,6 +2546,13 @@ int bt_iso_big_terminate(struct bt_iso_big *big)
 
 	if (IS_ENABLED(CONFIG_BT_ISO_BROADCASTER) &&
 	    bis->iso->iso.info.type == BT_ISO_CHAN_TYPE_BROADCASTER) {
+		/* Remove ISO data_path before stream destroyed */
+		SYS_SLIST_FOR_EACH_CONTAINER(&big->bis_channels, bis, node) {
+			bis->iso->err = BT_HCI_ERR_LOCALHOST_TERM_CONN;
+
+			bt_iso_remove_data_path(bis->iso);
+		}
+
```

Comment:
The purpose of this change is not found in the commit message. What is the purpose? 

From the core spec:
> The HCI_LE_Terminate_BIG command is used to terminate a BIG identified by the BIG_Handle parameter. The command also terminates the transmission of all BISes of the BIG, destroys the associated connection handles of the BISes in the BIG and removes the data paths for all BISes in the BIG.

and

> The HCI_LE_BIG_Terminate_Sync command is used to stop synchronizing or cancel the process of synchronizing to the BIG identified by the BIG_Handle parameter. The command also terminates the reception of BISes in the BIG specified in the HCI_LE_BIG_Create_Sync command, destroys the associated connection handles of the BISes in the BIG and removes the data paths for all BISes in the BIG.

Removing the data paths before sending the `HCI_LE_Terminate_BIG` or `HCI_LE_BIG_Terminate_Sync` commands is not needed, as the data paths shall be removed as part of the command. 

In fact, the only data path that survives a ISO disconnection, is CIS on the central side.

Reply:
Thank you for that. It will be removed.

The purpose of the change was:
"Remove ISO data path on host just prior to BIG termination while stream is defined."
In retrospect, that comment does not seem very helpful.

The change was motivated by an error message in samples/bluetooth/broadcast_audio_source
prior to starting the second broadcasting loop iteration:

```
<wrn> bt_hci_core: bt_hci_cmd_send_sync: opcode 0x206e status 0x0c
<err> bt_iso: bt_iso_connected: Unable to setup data path: -5
```

and by BLUETOOTH CORE SPECIFICATION Version 5.3 | Vol 4, Part E 7.8.109:

> If the Host issues this command more than once for the same Connection_
> Handle and direction before issuing the HCI_LE_Remove_ISO_Data_-
> Path command for that Connection_Handle and direction, the Controller shall
> return the error code Command Disallowed (0x0C).

I will look into that error further.

Reply:
> and by BLUETOOTH CORE SPECIFICATION Version 5.3 | Vol 4, Part E 7.8.109:
> 
> > If the Host issues this command more than once for the same Connection_
> > Handle and direction before issuing the HCI_LE_Remove_ISO_Data_-
> > Path command for that Connection_Handle and direction, the Controller shall
> > return the error code Command Disallowed (0x0C).
> 
> I will look into that error further.

Hmm, yeah that actually seem like a conflict in the specification, as the host technically never sends the HCI_LE_Remove_ISO_Data_Path command. Arguably, since this is called `setup` and not `add`, this entire paragraph doesn't really fit, as the name somewhat suggests that we are configuring a new or existing data path, and not _adding_ one. 

I would suggest that we raise an erratum for the BT spec to clarify this. 
@cvinayak Do you agree?
@carlst If Vinayak agrees, will you create the erratum? Otherwise I don't mind doing it.

Reply:
@Thalley @cvinayak A couple of things: I would prefer that a Zephyr developer handle communication with the BT SIG, but perhaps wait a bit until this PR is resolved as closely as possible. There are enough separate parts and details of the BT spec that I would like to (re)read to check if there really is an issue. Also, there may be PTS information (PTS test documentation and/or PTS tests) that shed further light.

I will continue resolving this PR (being a Zephyr user and not a Zephyr developer, my comfort zone is currently more in reporting Issues than implementing PRs so I appreciate your continued patience).

Reply:
@Thalley  and @carlst I have tried to test this PR last week, and did not come across the requirement to change implementation in Host. I did not face the said command disallowed reason as I did not try to terminate the BIG. I am assume the said issue would be occurring after terminate BIG and to try to setup a new data path. But, for the terminate BIG we have, in BT Spec v5.3, Vol. 4, Part E, section 7.8.105 LE Terminate BIG command

```
The HCI_LE_Terminate_BIG command is used to terminate a BIG identified by
the BIG_Handle parameter. The command also terminates the transmission of
all BISes of the BIG, destroys the associated connection handles of the BISes
in the BIG and removes the data paths for all BISes in the BIG.
```

The controller is responsible to remove the data path implicitly if not already when a BIG is terminated. It is fine for the Host to not issue LE Remove ISO Data Path command if it is terminating a BIG using LE Terminate BIG command.

Reply:
> The controller is responsible to remove the data path implicitly if not already when a BIG is terminated. It is fine for the Host to not issue LE Remove ISO Data Path command if it is terminating a BIG using LE Terminate BIG command.
> 
> At this point, I do not see an erratum is required.

I agree that it shouldn't be necessary for the host to remove it, as it is, as you said, implicitly removed. However, IMO, that is not covered by

> If the Host issues this command more than once for the same Connection_Handle and direction before issuing the HCI_LE_Remove_ISO_Data_Path command for that Connection_Handle and direction, the Controller shall return the error code Command Disallowed (0x0C).

Because the host in that scenario never issues the `HCI_LE_Remove_ISO_Data_Path`. It won't be a technical erratum, if any, but rather just a semantic one. It won't change anything besides making it more clear that `HCI_LE_Setup_ISO_Data_Path` can be sent for the same Connection_handle and direction, if the BIG has been terminated (which implicitly removes the data paths).

Reply:
> It won't change anything besides making it more clear that HCI_LE_Setup_ISO_Data_Path can be sent for the same Connection_handle and direction, if the BIG has been terminated (which implicitly removes the data paths).

I will raise a Bluetooth SIG Jira request once I get the access permissions to login.